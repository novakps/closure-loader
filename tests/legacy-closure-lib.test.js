const path = require('path');
const webpack = require('webpack');
const execBundle = require('./helpers/execBundle');

// these tests are slower, because they map all the deps in google-closure-library
jest.setTimeout(40000);

const config = {
    module: {
        rules: [
            {
                test: /google-closure-library\/closure\/goog\/base/,
                use: [
                    'imports-loader?this=>{goog:{}}&goog=>this.goog',
                    'exports-loader?goog',
                ],
            },
            {
                test: /.*\.js/,
                exclude: [/google-closure-library\/closure\/goog\/base\.js$/],
                use: [
                    {
                        loader: require.resolve('..'),
                        options: {
                            paths: [
                                path.join(
                                    __dirname,
                                    '..',
                                    'node_modules/google-closure-library/closure/goog',
                                ),
                            ],
                            es6mode: false,
                            watch: false,
                        },
                    },
                ],
            },
        ],
    },
    plugins: [
        new webpack.ProvidePlugin({
            goog: 'google-closure-library/closure/goog/base',
        }),
    ],
};

test('loads closure library modules with goog.require()', async () => {
    const result = await execBundle('legacy-closure-lib', config);
    expect(result.code).toBe(0);
    expect(result.stdout).toMatchSnapshot();
});
