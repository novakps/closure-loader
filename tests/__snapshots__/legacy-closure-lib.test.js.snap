// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads closure library modules with goog.require() 1`] = `
"Inner box in outer box true
Outer box in inner box false
calculating if one box contains another
Outer box in inner box false"
`;
